src/goto-synthesizer/synthesizer_utils.cpp

```diff
@@ -112,3 +116,55 @@ void annotate_invariants(
       invariant_map_entry.second;
   }
 }
+
+bool is_instruction_in_transfomed_loop(
+  const loop_idt &loop_id,
+  const goto_functiont &function,
+  unsigned location_number_of_target)
+{
+  // The loop body after transformation are instructions from
+  // loop havocing instructions
+  // to
+  // loop end of transformed code.
+
+  bool result = false;
+  unsigned location_number_of_havocing = UINT_MAX;
+  for(goto_programt::const_targett it = function.body.instructions.begin();
+      it != function.body.instructions.end();
+      ++it)
+  {
+    if(is_loop_havoc(*it) && it->loop_number == loop_id.loop_number)
+    {
+      location_number_of_havocing = it->location_number;
+    }
+
+    if(is_transformed_loop_end(it) && it->loop_number == loop_id.loop_number)
+    {
+      result =
+        result || (location_number_of_havocing < location_number_of_target &&
+                   location_number_of_target < it->location_number);
```

Comment:
Can't you just do `if(....) return true;`? Once `result` has become `true`, the remainder of the expression will never be evaluated again.